server/controllers/bookController.js

```diff
@@ -90,6 +90,50 @@ class BookController {
       Utils.errorStatus(res, 500, error.message);
     }
   }
+
+  static async lendBook(req, res) {
+    const { isbn, title, patronId, cost} = req.body;
+    
+    try {
+      const booksNotReturned = await models.Borrowed_books.findAndCountAll({ where: { patronId, returned : false } });
+
+      if (booksNotReturned.count >= 3) {
+        return Utils.errorStatus(res, 405, 'You cannot have more than 3 books in your possession');        
+      } 
+
+      const today = new Date();
+      const overdueBook = booksNotReturned.rows.find((book) => book.dueDate < today);
+
+      if(overdueBook) {
+        const dueDate = overdueBook.dueDate.toString().substring(0, 15);
+        const responseObj = {
+          message: `The book titled ${overdueBook.title.toUpperCase()} was due for return on ${dueDate}. Kindly return before we can proceed with a new request`
```

Comment:
When the patron has 2 books that are overdue, it only returns error message that the first book is overdue. Find a way to loop through it if the overdue books are 2... If it is three or more the first validation `You cannot have more than 3 books in your possession` would cover that.

Reply:
This too.